syntax = "proto3";

package qa_service;

// The question answering service definition
service QuestionAnswerer {
  // Send a question and receive an answer
  rpc AskQuestion (QuestionRequest) returns (AnswerResponse) {}
}

// The request message containing the question
message QuestionRequest {
  string question = 1;  // The question text
  repeated string context_tables = 2;  // Optional: specific tables to consider
}

// The response message containing the answer
message AnswerResponse {
  string answer = 1;  // The generated answer
  bool success = 2;  // Whether the operation was successful
  string error = 3;  // Error message if any
  float confidence = 4;  // Confidence score of the answer
} 